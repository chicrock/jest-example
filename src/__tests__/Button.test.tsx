import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import sinon from "sinon";
import Button from "../Components/Button";

configure({ adapter: new Adapter() });

describe("<Button />", () => {
  it("renders a <Button /> components", () => {
    const button = shallow(<Button value="Test" />);
    expect(button.length).toEqual(1);
  });

  it("renders with props", () => {
    const button = shallow(<Button value="Test" className={"asdf"} />);
    expect(button.find(".asdf").length).toEqual(1);
    expect(button.props().className).toEqual("asdf");
    expect(button.props().value).toEqual("Test");
  });

  it("renders to Button click Event", () => {
    const onClick = sinon.spy();
    const button = shallow(<Button value="Test" onClick={onClick} />);
    button.simulate("click");
    expect(onClick).toHaveProperty("callCount", 1);
  });

  it("renders the output", () => {
    const button = shallow(<Button value="Test" />);
    const expectedOutput = '<input type="submit" value="Test"';

    const realOutput = button.html();
    expect(realOutput).toContain(expectedOutput);
    // expect(realOutput.indexOf(expectedOutput) > -1).toEqual(true);
  });

  it("renders the value", () => {
    const button = shallow(<Button value="Test" />);

    expect(button.html()).toContain("Test");
  });
});
