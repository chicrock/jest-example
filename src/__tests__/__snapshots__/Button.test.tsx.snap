// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test with enzyme serializer 1`] = `
<styled.input
  disabled={false}
  type="submit"
  value="Test"
/>
`;

exports[`<Button /> test with enzyme serializer 2`] = `
<styled.input
  disabled={false}
  type="submit"
  value="Test12"
/>
`;

exports[`<Button /> test with enzyme serializer 3`] = `
<styled.input
  disabled={false}
  type="submit"
  value="Values"
/>
`;

exports[`<Button /> test with snapshot 1`] = `
<input
  className="sc-bdVaJa didhXZ"
  disabled={false}
  type="submit"
  value="Test"
/>
`;
